### PR TITLE
Upgrade tests split

### PR DESCRIPTION
**Pull request by xpivarc**
**What this PR does / why we need it**:
Operator tests up until now were a mixed set of install, upgrade, and configuration tests in one huge file. This has a few downsides:
1. We always perform reinstallation and other unnecessary actions (see Before/After each)
2. We can't execute only a set of tests.
3. There is too much noise

This PR aims to execute upgrade and configuration tests in separate lanes. This is useful to test operator tests on Kubernetes with PSA enabled.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: tests/operator/operator_test.go [hunk 1]
```diff
@@ -678,342 +667,7 @@ var _ = Describe("[Serial][sig-operator]Operator", Serial, decorators.SigOperato
...
-			deleteAllVMIs(migratableVMIs)
-
-			By("Deleting KubeVirt object")
-			deleteAllKvAndWait(false)
-		},
-			Entry("by patching KubeVirt CR", false),
-			Entry("by updating virt-operator", true),
-		)
-	})
+	updateTest()
```
Comment (acardace) [inline tests/operator/operator_test.go:670]: Why this approach? Can't we have 2 separated files (without the first one calling into the second one)

Comment (xpivarc) [reply tests/operator/operator_test.go:670]: My first attempt to split it totally resulted in not functioning tests. Reason being the dependencies between variables, before/after blocks and tests seems to be too much and therefore require a more iterative approach to this. I will iterate a few times over this with upcoming PRs and I think we will get there. Sounds good?

Comment (acardace) [reply tests/operator/operator_test.go:670]: Gotcha, no problem with that.

File: automation/test.sh [hunk 2]
```diff
@@ -377,7 +377,15 @@ if [[ -z ${KUBEVIRT_E2E_FOCUS} && -z ${KUBEVIRT_E2E_SKIP} ]]; then
   elif [[ $TARGET =~ sig-monitoring ]]; then
     label_filter='(sig-monitoring)'
   elif [[ $TARGET =~ sig-operator ]]; then
-    label_filter='(sig-operator)'
+    if ! [[ $TARGET =~ sig-operator-upgrade ]]; then
```
Comment (dhiller) [inline automation/test.sh:380]: I am a bit confused by this logic, is it the same as this one?

```sh
    if [[ $TARGET =~ sig-operator-upgrade ]]; then
      label_filter='(sig-operator && Upgrade)'
    elif [[ $TARGET =~ sig-operator-configuration ]]; then
      label_filter='(sig-operator && !(Upgrade))'
    else
      label_filter='(sig-operator)'
    fi
```

File: tests/decorators/decorators.go [hunk 3]
```diff
@@ -3,28 +3,34 @@ package decorators
...
-	SRIOV                = []interface{}{Label("SRIOV")}
-	NonRoot              = []interface{}{Label("verify-non-root")}
-	NativeSsh            = []interface{}{Label("native-ssh")}
-	ExcludeNativeSsh     = []interface{}{Label("exclude-native-ssh")}
-	Reenlightenment      = []interface{}{Label("Reenlightenment")}
-	Invtsc               = []interface{}{Label("Invtsc")}
-	PasstGate            = []interface{}{Label("PasstGate")}
+
+	// HW
+	GPU        = []interface{}{Label("GPU")}
```
Comment (dhiller) [inline tests/decorators/decorators.go:16]: IIUC it is redundant to wrap labels into an array, could we remove those?

File: tests/operator/operator_test.go [hunk 4]
```diff
@@ -91,950 +74,43 @@ import (
...
+	var aggregatorClient *aggregatorclient.Clientset
 
-			supportedVersions := []string{}
-			for _, version := range crd.Spec.Versions {
-				supportedVersions = append(supportedVersions, version.Name)
-			}
+	tests.DeprecatedBeforeAll(func() {
+		virtClient = kubevirt.Client()
+		config, err := kubecli.GetKubevirtClientConfig()
+		util2.PanicOnError(err)
```
Comment (dhiller) [inline tests/operator/operator_test.go:87]:
```suggestion
		testutils.PanicOnError(err)
```

File: tests/operator/upgrade_test.go [hunk 5]
```diff
@@ -0,0 +1,478 @@
...
+	"kubevirt.io/kubevirt/tests/decorators"
+	"kubevirt.io/kubevirt/tests/flags"
+	"kubevirt.io/kubevirt/tests/framework/checks"
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	. "kubevirt.io/kubevirt/tests/framework/matcher"
+	"kubevirt.io/kubevirt/tests/libstorage"
+	"kubevirt.io/kubevirt/tests/testsuite"
+)
+
+func updateTest() {
```
Comment (dhiller) [inline tests/operator/upgrade_test.go:39]: It's confusing to read the filename as `upgrade_test` and then find a function inside being called `updateTest()`, maybe rename file to `update_test.go`?

**Review by dhiller — COMMENTED** [review]
Looks good already, some questions inline.

File: tests/operator/operator_test.go (hunk 4 above)
Comment (xpivarc) [reply tests/operator/operator_test.go:87]: I just moved this so I would prefer to do this in follow up

File: tests/operator/upgrade_test.go (hunk 5 above)
Comment (xpivarc) [reply tests/operator/upgrade_test.go]: Synced

File: tests/decorators/decorators.go (hunk 3 above)
Comment (xpivarc) [reply tests/decorators/decorators.go:16]: I think we still need some documentation around this and so on so I would leave it for another time.

File: automation/test.sh (hunk 2 above)
Comment (xpivarc) [reply automation/test.sh]: Done

**Comment by xpivarc** [comment]
@acardace I have to give up on splitting the test into separate files.

**Comment by acardace** [comment]
> @acardace I have to give up on splitting the test into separate files.

Fine to me, what's important is to split them.

**Comment by xpivarc** [comment]
@dhiller PTAL

**Review by dhiller — APPROVED** [review]

**Pull request merged**